File: go-sdk/pkg/models/operations/demoteabranch.go
package operations

type DemoteABranchPathParams struct {
	Database     float64 `pathParam:"style=simple,explode=false,name=database"`
	Name         float64 `pathParam:"style=simple,explode=false,name=name"`
	Organization float64 `pathParam:"style=simple,explode=false,name=organization"`
}

type DemoteABranchRequest struct {
	PathParams DemoteABranchPathParams
}

type DemoteABranchResponse struct {
	ContentType                           string
	StatusCode                            int64
	DemoteABranch200ApplicationJSONObject map[string]map[string]interface{}
}

File: ts-sdk/src/sdk/models/operations/demoteabranch.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Type } from "class-transformer";


export class DemoteABranchPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=database" })
  database: number;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=name" })
  name: number;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: number;
}

export class DemoteABranchRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: DemoteABranchPathParams;
}

export class DemoteABranchResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  demoteABranch200ApplicationJSONObject?: Record<string, Record<string, any>>;
}